### PR TITLE
Added check to ensure 'password' is defined in options before using it. password is optional when using pub/private keys

### PR DESCRIPTION
**Pull request by J-Dill**
When you use the WP_Filesystem_SSH2 class along with public_key/private_key authentication, you get the following warning message in your debug logs:  
```
“Undefined array key “password““ at `./wp-admin/includes/class-wp-filesystem-ssh2.php:153`
```
Providing a password with public/private key authentication is optional, but the code currently expects there to always be a password provided. This PR adds a null coalescing check to see if a password is defined in the options. If it is not, then it will define null for the parameter as before.

Trac ticket: https://core.trac.wordpress.org/ticket/33196

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by costdev — APPROVED** [review]
Thanks for the PR @J-Dill! LGTM and tests well :+1:

File: src/wp-admin/includes/class-wp-filesystem-ssh2.php [hunk 1]
```diff
@@ -150,7 +150,8 @@ public function connect() {
 				return false;
 			}
 		} else {
-			if ( ! @ssh2_auth_pubkey_file( $this->link, $this->options['username'], $this->options['public_key'], $this->options['private_key'], $this->options['password'] ) ) {
+			$password = isset( $this->options['password'] ) ? $this->options['password'] : null;
```
Comment (costdev) [inline src/wp-admin/includes/class-wp-filesystem-ssh2.php]: On second thought, we can just set `$this->options['password'] = null;` in an `else` block [here](https://github.com/WordPress/wordpress-develop/blob/e61d17a5ed156c57104f214846c25224131f49a4/src/wp-admin/includes/class-wp-filesystem-ssh2.php#L104-L106).

So:
```php
if ( ! $this->keys ) {
	$this->errors->add( 'empty_password', __( 'SSH2 password is required' ) );
} else {
	$this->options['password'] = null;
}
```
@J-Dill Are you happy to make this change to the PR?

Comment (J-Dill) [reply src/wp-admin/includes/class-wp-filesystem-ssh2.php]: Yes that seems like a reasonable way to do it as well. I will make that change here shortly.

Comment (J-Dill) [reply src/wp-admin/includes/class-wp-filesystem-ssh2.php]: @costdev Done. Please take another look.

Comment (costdev) [reply src/wp-admin/includes/class-wp-filesystem-ssh2.php]: Thanks! That looks good to me!

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in [r56111](https://core.trac.wordpress.org/changeset/56111).

**Comment by sabernhardt** [comment]
@J-Dill If you have a moment, could you connect your GitHub account to your wordpress.org one? This change is slated to be included in the upcoming WordPress 6.3 release on August 8th, and it would be great if we could give you attribution by mentioning you on the credits screen. You can find more details on how contributions are tracked and how connecting GitHub accounts helps in [this post on the Make WordPress Core blog](https://make.wordpress.org/core/2020/03/19/associating-github-accounts-with-wordpress-org-profiles/).

**Comment by J-Dill** [comment]
@sabernhardt Thanks for the notification, I have created an account in wordpress.org and linked it!

**Comment by sabernhardt** [comment]
Sorry. We found '[lowlydev](https://profiles.wordpress.org/lowlydev/)' from your email, but that does not show a connection to GitHub. Do you have a more appropriate WordPress.org username?

**Comment by J-Dill** [comment]
Ah yes that account also is right. I linked my GitHub to that account as well.

**Pull request closed without merging**